fluent-syntax/src/parser/ftlstream.rs

```diff
@@ -148,7 +148,7 @@ impl<'p> ParserStream<'p> {
     }
 
     pub fn get_slice(&self, start: usize, end: usize) -> &'p str {
-        str::from_utf8(&self.source[start..end]).expect("Slicing the source failed")
+        unsafe { str::from_utf8_unchecked(&self.source[start..end]) }
```

Comment:
This function should be unsafe. The start or end might not be at the start or end of a character, which makes the str invalid.

Reply:
@bjorn3 - please, comment in #82 where we're discussing it. In particular, I made a claim that the start and end must be at the start and end of a character due to how the parser works.